appserver/web/web-core/src/main/java/org/apache/catalina/servlets/CGIServlet.java

```diff
@@ -1692,7 +1693,7 @@ protected void run() throws IOException {
 
             //create query arguments
             StringBuilder cmdAndArgs = new StringBuilder();
-            if (command.indexOf(" ") < 0) {
+            if (command.indexOf(' ') == -1) {
```

Comment:
I have never heard of something other than `-1` being returned, the documentation for `String` says  `if no such character occurs in this string, then -1 is returned`.

Reply:
Its not a common thing in Java I guess but there might be the odd API doing that. String isn't. I was just wondering as I have seen lots of `< 0` to do this and far less `== -1`.

Reply:
One example In Java is https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#binarySearch(java.lang.Object[],%20java.lang.Object)

Reply:
Comparing < is generally faster than == (comparing bits and searching first difference). But on current machines it is only a premature optimization, because both are single cpu tact and the most expensive opration is moving data from/to memory right in time to be processed.
So the most important thing is readibility, but ... yet one thing. indexOf(' ') takes int as argument, not char, and it has to manage UTF-16 characters, so it even is not faster.
It tried to test it, but results are not provable, seems rather similar for all three variants (contains(" ") is third).